src/components/AssignmentCreatorForm.jsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import {t} from 'i18next';
+import classnames from 'classnames';
+import chrono from 'chrono-node';
+import PropTypes from 'prop-types';
+import ImmutablePropTypes from 'react-immutable-proptypes';
+import {Field, reduxForm} from 'redux-form/immutable';
+
+import AssignmentCreatorTextField from './AssignmentCreatorTextField';
+import AssignmentCreatorSelectField from './AssignmentCreatorSelectField';
+
+function validate(values) {
+  const errors = {};
+  const course = values.get('course');
+  const date = values.get('date');
+
+  if (!course) {
+    errors.course = t('assignment-creator.no-class-selected');
+  } else if (course === '') {
+    errors.course = t('assignment-creator.no-class-selected');
+  }
+  if (!date) {
+    errors.date = t('assignment-creator.date-not-valid');
+  } else if (!date.parsedDate) {
+    errors.date = t('assignment-creator.date-not-valid');
+  } else if (date.parsedDate < Date.now()) {
+    errors.date = t('assignment-creator.past-date-not-valid');
+  }
+  return errors;
+}
+
+function parseDate(value) {
+  const parsedResponse = chrono.parse(value);
+  let parsedDate;
+  if (parsedResponse[0]) {
+    parsedDate = parsedResponse[0].start.date();
+  }
+  return {
+    string: value,
+    parsedDate,
+  };
```

Comment:
You mentioned that this didn’t work right when you used an `Immutable.Record`—can you elaborate? Or push a version that does that so I can poke around with it?